collect_app/src/main/java/org/odk/collect/android/widgets/WidgetFactory.java

```diff
@@ -229,9 +232,16 @@ public static QuestionWidget createWidgetFromPrompt(FormEntryPrompt fep, Context
                     case Constants.DATATYPE_DECIMAL:
                         questionWidget = new RangeDecimalWidget(context, fep);
                         break;
+                    default:
+                        questionWidget = new StringWidget(context, fep, readOnlyOverride);
```

Comment:
Can't we assume `questionWidget = new StringWidget(context, fep, readOnlyOverride);` at the beginning instead of using many defaults where in all cases we use the same?

Reply:
See the conversation above!

Reply:
Other than that problem, @grzesiek2010, I do love that you wanted to remove the duplication. I might have done the same.

Reply:
It's kind of subtle. I got tripped up too because I kept thinking about the release of the `StringWidget` itself but the whole point is that the problem is with its **resources**, not itself.

Reply:
Yes, I sometimes miss C++ destructors.